jupyterlab/debuglog.py

```diff
@@ -0,0 +1,66 @@
+# coding: utf-8
+"""A mixin for adding a debug log file.
+
+"""
+
+# Copyright (c) Jupyter Development Team.
+# Distributed under the terms of the Modified BSD License.
+
+import contextlib
+import logging
+import os
+import sys
+import tempfile
+import traceback
+
+from traitlets import HasTraits, Unicode, default
+
+class DebugLogFileMixin(HasTraits):
+    debug_log_path = Unicode('', config=True, help='TODO')
+
+    @contextlib.contextmanager
+    def debug_logging(self):
+        log_path = self.debug_log_path
+        if not log_path:
+            handle, log_path = tempfile.mkstemp(prefix='jupyterlab-log-', suffix='.log')
```

Comment:
Why not create a jupyterlab-debug file in the cwd by default, same as npm?

Reply:
I think npm puts it in its config dir now. I always considered getting the log files in cwd a nuisance. Having the tmp / auto-name also ensure that we can have multiple commands running at the same time, and you will avoid permission issues and races on multi-user systems. It also ensures some level of automatic cleanup.

Reply:
Clarification, npm puts it in its cache dir (`npm config get cache`).